packages/commons-test/test/makeApiProblem.unit.test.ts

```diff
@@ -0,0 +1,406 @@
+import { constants } from "http2";
+import {
+  ApiError,
+  CommonErrorCodes,
+  Problem,
+  badRequestError,
+  commonErrorCodes,
+  emptyErrorMapper,
+  featureFlagNotEnabled,
+  makeApiProblemBuilder,
+  serviceErrorCode,
+  tokenVerificationFailed,
+  tooManyRequestsError,
+  unauthorizedError,
+} from "pagopa-interop-models";
+import { match } from "ts-pattern";
+import { describe, expect, it, vi } from "vitest";
+import { z } from "zod";
+import { getMockContext } from "../src/testUtils.js";
+
+export const testErrorCodes = {
+  testError1: "0001",
+  testError2: "0002",
+  testError3: "0003",
+};
+
+export type TestErrorCodes = keyof typeof testErrorCodes;
+
+export function testError1(): ApiError<TestErrorCodes> {
+  return new ApiError({
+    detail: "This is a test error 1",
+    code: "testError1",
+    title: "Test Error 1",
+  });
+}
+
+export function testError2(): ApiError<TestErrorCodes> {
+  return new ApiError({
+    detail: "This is a test error 2",
+    code: "testError2",
+    title: "Test Error 2",
+  });
+}
+
+export function testError3(): ApiError<TestErrorCodes> {
+  return new ApiError({
+    detail: "This is a test error 3",
+    code: "testError3",
+    title: "Test Error 3",
+  });
+}
```

Comment:
Considering that their definition works in tandem with `defaultTestErrorMapper` ,
for readability reason those methods could be renamed to (or something more auto explicative)

```suggestion
export function expectedBadRequestError(): ApiError<TestErrorCodes> {
  return new ApiError({
    detail: "This is a test error 1",
    code: "testError1",
    title: "Test Error 1",
  });
}

export function expectedNotFoundError(): ApiError<TestErrorCodes> {
  return new ApiError({
    detail: "This is a test error 2",
    code: "testError2",
    title: "Test Error 2",
  });
}

export function expectedInternalError(): ApiError<TestErrorCodes> {
  return new ApiError({
    detail: "This is a test error 3",
    code: "testError3",
    title: "Test Error 3",
  });
}
```